certbot-nginx/certbot_nginx/_internal/http_01.py

```diff
@@ -131,21 +134,22 @@ def _default_listen_addresses(self):
             if not ipv6only:
                 # If ipv6only=on is not already present in the config
                 ipv6_addr = ipv6_addr + " ipv6only=on"
-            addresses = [obj.Addr.fromstring(default_addr),
-                         obj.Addr.fromstring(ipv6_addr)]
+            addresses = [Addr.fromstring(default_addr),
+                         Addr.fromstring(ipv6_addr)]
             logger.debug(("Using default addresses %s and %s for authentication."),
                         default_addr,
                         ipv6_addr)
         else:
-            addresses = [obj.Addr.fromstring(default_addr)]
+            addresses = [Addr.fromstring(default_addr)]
             logger.debug("Using default address %s for authentication.",
                         default_addr)
-        return addresses
 
-    def _get_validation_path(self, achall):
+        return [address for address in addresses if address]
```

Comment:
As a consequence of the above, we want to clear the `None` elements in `addresses` before we return from `_default_listen_addresses()`.

Reply:
Basically here mypy uncovered a flaw: without appropriate nullity checks, `default_listen_addresses()` may return a list containing `None` elements, that could break `_make_server_block()` because we call `addr.to_string(include_default=False)` to each `addr` generated in the list returned by `default_listen_addresses()`.